src/hotspot/os/windows/os_windows.cpp

```diff
@@ -300,15 +302,27 @@ void os::init_system_properties_values() {
       home_dir[MAX_PATH] = '\0';
     } else {
       os::jvm_path(home_dir, sizeof(home_dir));
-      // Found the full path to jvm.dll.
-      // Now cut the path to <java_home>/jre if we can.
-      *(strrchr(home_dir, '\\')) = '\0';  // get rid of \jvm.dll
+      // Found the full path to the binary. It is normally of this structure:
+      //   <jdk_path>/bin/<hotspot_variant>/jvm.dll
+      // but can also be like this for a statically linked binary:
+      //   <jdk_path>/bin/<executable>.exe
       pslash = strrchr(home_dir, '\\');
       if (pslash != nullptr) {
-        *pslash = '\0';                   // get rid of \{client|server}
+        if (strncmp(pslash + 1, JVM_LIB_NAME, strlen(JVM_LIB_NAME)) == 0) {
+          // Binary name is jvm.dll. Get rid of \jvm.dll.
+          *pslash = '\0';
+        }
+
+        // Get rid of \hotspot_variant>, if binary is libjvm.so,
```

Comment:
libjvm.so -> jvm.dll?

Reply:
Good catch. I copied and adapted this code from the unix solution, but missed this. Fixed.